### ISSUE TITLE
add heater & home behavior analysis and heating anticipation

### ISSUE THREAD

**Issue by PaulSchryve** [mskg/homey-heating#152]
**Is your feature request related to a problem? Please describe.**
I use electric heaters with thermal inertia.  With those systems, the temperature grows when power is up, but also after it has been cut due to inertia. 
Current programming mode is not able to manage this => final temperature is not as requested.

Another point is that delays to increase temperature can be long, so all schedules have to be anticipated a lot. So if by a flow you activate a mode with a timing, it's really complicated to anticipate what will be the real temperature. (for example, in case you have different schedules for your rooms, for some rooms you need to anticipate a lot due to high temperature difference, for some other not).

**Describe the solution you'd like**

I already have had a system able to manage this in the past. (Netatmo)
1/ first, the system has to learn the behavior of the devices. During a period of time, it learns the influence of heater activation on the temperature of the room, and inertia.
(for example, the systems can know that to go from 12 to 20 degrees, it tooks 2 hours of activation, with more details on différents configurations - starting temperature / final one; and that when power is off the heater continues to heat for a couple of minutes / hours.)
2/ After a Learning phase, the system is able to anticipate activation of the heaters to reach the target tempetature at the time it is scheduled.
(for example, at 10h you want to get 20 degrees. At 7h the temperature is 12 degrees. The system then monitors temperature in the room, knows that to go from 12 to 20 it needs 2 hours of heating, then at 10-2 = 8h it starts the heater, so that at 10h the real temperature will be 20 degrees.)
3/ The system is also able to stop the heater even if the target temperature is not reached, as it knows from Learning phase that with 2 hours of activation, the final temperature will be reached with inertia.

**Describe alternatives you've considered**
Anticipate all schedules => in that case i can have the target temperature or more. Sometimes more is too much. (for example, if the initial temperature was very low, the heater will activate a long time and get a lot of inertia, then it will heat a lot more, temperature will be really to high)

**Additional context**
This features are usefull for all inertia systems, heating floor for example.
The fact to anticipate heating is a big plus, you will have to implement your schedules not taking care of the system, but only with your real needs.

It's probably a big change, but it will add a lot of value on the system.
For Netatmo they call this feature auto adapt 
https://www.netatmo.com/en-eu/energy/thermostat

Do not hesitate if you want more details.

**Comment by mskg** [comment]
I should take a lot of money for that feature :)

You have a lot degrees of freedom you have to take into consideration: Temperature of neighbor rooms, outside temperature, starting point, target temperature, other heating jobs running parallel and their ...

**Comment by PaulSchryve** [comment]
You're right, a lot of things can influence the result, but from my past use of netatmo solution, this was possible & efficient. I do'nt know how they solve the difficulty, perhaps a simple model is already a big progress. (example during 1 month you monitor time to change temperature, and use a medium as a rule) No perfect but probably efficient, with perhaps an option to adjust the parameter.

**Comment by PaulSchryve** [comment]
some other thermostat implementation seem to be available, from vera lite platform / seems to be the base of some really complete jeedom plugin  https://github.com/AntorFr/SmartVT
